src/result_gen.py

```diff
@@ -0,0 +1,563 @@
+import os
```

Comment:
This module is very large. Please consider splitting this up into submodules. 
Could we structure this along the different results? e.g.
- `height_results.py`
- `blur_results.py`

Reply:
TBD later in next PR